### PR TITLE
Add globalization

### PR DESCRIPTION
**Pull request by deepakrkris**
initial checkin for strong globalization

### CODE REVIEW

File: lib/rest-adapter.js [hunk 1]
```diff
@@ -331,8 +335,7 @@ RestAdapter.prototype._shouldHandleErrors = function() {
 RestAdapter.remoteMethodNotFoundHandler = function(className) {
   className = className || '(unknown)';
   return function restRemoteMethodNotFound(req, res, next) {
-    var message = 'Shared class "' + className + '"' +
-      ' has no method handling ' + req.method + ' ' + req.url;
+    var message = g.f('Shared class %s has no method handling %s %s',className, req.method, req.url);
```
Comment (Setogit) [inline lib/rest-adapter.js]: space char is missing before `className`

File: lib/shared-method.js [hunk 2]
```diff
@@ -444,7 +450,7 @@ function convertValueToTargetType(argName, value, targetType) {
         case 'object':
           return value;
         default:
-          throw new badArgumentError(argName + ' must be ' + targetType);
+          throw new badArgumentError(argName + g.f(' must be ') + targetType);
```
Comment (Setogit) [inline lib/shared-method.js]: globalization bug: use g.f('%s must be %s.', argName, targetType)

File: lib/shared-method.js [hunk 3]
```diff
@@ -571,9 +577,9 @@ SharedMethod.toResult = function(returns, raw, ctx) {
     var name = item.name || item.arg;
     var targetType = convertToBasicRemotingType(item.type);
     if (targetType === 'file') {
-      console.warn('%s: discarded non-root return argument %s of type "file"',
+      console.warn(g.f('%s: discarded non-root return argument %s of type "file"',
```
Comment (Setogit) [inline lib/shared-method.js]: {{"file"}} ?

File: package.json [hunk 4]
```diff
@@ -33,7 +33,8 @@
     "sse": "0.0.6",
     "strong-error-handler": "^1.0.0",
     "traverse": "^0.6.6",
-    "xml2js": "^0.4.8"
+    "xml2js": "^0.4.8",
+    "strong-globalize": "^2.6.2"
```
Comment (Setogit) [inline package.json]: "^2.x"

**Comment by Setogit** [comment]
@deepakrkris commented

**Comment by Setogit** [comment]
Pls fix the 4 lint errors on travis CI:
```
/home/travis/build/strongloop/strong-remoting/lib/http-context.js
  664:61  error  A space is required after ','  comma-spacing
/home/travis/build/strongloop/strong-remoting/lib/jsonrpc-adapter.js
  90:63  error  Missing trailing comma  comma-dangle
/home/travis/build/strongloop/strong-remoting/lib/rest-adapter.js
  337:1   error  Line 337 exceeds the maximum line length of 90  max-len
  337:69  error  A space is required after ','                   comma-spacing

```

File: lib/http-context.js [hunk 5]
```diff
@@ -622,11 +624,11 @@ HttpContext.prototype.done = function(cb) {
         });
         // TODO(ritch) support multi-part streams
       } else {
-        cb(new Error('unsupported stream type: ' + returnStreamDesc.type));
+        cb(new Error(g.f('unsupported stream type: ') + returnStreamDesc.type));
```
Comment (0candy) [inline lib/http-context.js]: Should be `g.f('unsupported stream type: %s', returnStreamDesc.type)`

File: lib/shared-method.js [hunk 6]
```diff
@@ -366,14 +370,14 @@ function coerceAccepts(uarg, desc) {
   // FIXME(bajtos) "null" should be treated as no value too
   if (actualType === 'undefined') {
     if (desc.required) {
-      throw new badArgumentError(name + ' is a required arg');
+      throw new badArgumentError(name + g.f(' is a required arg'));
```
Comment (0candy) [inline lib/shared-method.js]: Should be `g.f('%s is a required arg', name)`

File: lib/shared-method.js [hunk 7]
```diff
@@ -366,14 +370,14 @@ function coerceAccepts(uarg, desc) {
...
     } else {
       return undefined;
     }
   }
 
   if (actualType === 'number' && Number.isNaN(uarg)) {
-    throw new badArgumentError(name + ' must be a number');
+    throw new badArgumentError(name + g.f(' must be a number'));
```
Comment (0candy) [inline lib/shared-method.js]: Similar to above

File: intl/en/messages.json [hunk 8]
```diff
@@ -0,0 +1,3 @@
+{
+  "22b8f1a3c27ad6df47f37a733dc8bd7f": "method is required when calling invoke()"
+}
```
Comment (davidcheung) [inline intl/en/messages.json]: please re-run `slt-globalize -e` to get the latest strings every time you make change to the translatable strings

File: package.json (hunk 4 above)
Comment (davidcheung) [inline package.json]: would prefer to see it alphabetical order 
`npm install --save strong-globalize` would make it alphabetical order

File: lib/socket-io-context.js [hunk 9]
```diff
@@ -19,6 +19,11 @@ var util = require('util');
 var inherits = util.inherits;
 var assert = require('assert');
 
+
+var g = require('strong-globalize')();
+
+
+
```
Comment (davidcheung) [inline lib/socket-io-context.js]: please pad with only 1 line of blankspace

File: index.js [hunk 10]
```diff
@@ -7,5 +7,9 @@
  * remotes ~ public api
  */
 
+var SG = require('strong-globalize');
+SG.SetRootDir(__dirname);
+var g = SG();
```
Comment (deepakrkris) [inline index.js]: var g = SG(); needs to be removed since it is not used in the script

**Comment by superkhau** [comment]
LGTM

**Pull request merged**